### PR TITLE
Use Brakeman in all Rails projects

### PR DESCRIPTION
**Pull request by lozette**
Proposal to use Brakeman security analysis tool in all Rails projects https://brakemanscanner.org/

### CODE REVIEW

File: rfc-024-use-brakeman.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+## Problem
+
+Security is a key principle of dxw's development. dxw projects are subject to a 
+penetration test before entering the live stage of their lifecycle. It is better 
+to address security vulnerabilities in the development  & testing stage, before 
+the code goes to production. Fixing vulnerabilities in live code can be costly
+in terms of time and complexity.
+
+Developers should learn best security practises and common vulnerabilities in Rails
```
Comment (pezholio) [inline rfc-024-use-brakeman.md:20]:
```suggestion
Developers should learn best security practices and common vulnerabilities in Rails
```

Comment (lozette) [reply rfc-024-use-brakeman.md]: Argh! Thanks!

**Comment by lozette** [comment]
Actually having looked at the Brakeman docs, it does not encourage running Brakeman in your default rake task: https://brakemanscanner.org/docs/rake/

So I'm going to remove that bit of the RFC.

File: rfc-024-use-brakeman.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+[Brakeman](https://brakemanscanner.org/) is a static security scanning tool for
+use on Rails projects. 
+
+Security is a key principle in all of dxw's projects. Using Brakeman during all
+stages of development & testing ensures we can be more confident of addressing
+and eliminating vulnerabilities before our code is deployed to production.
+
+## Problem
+
+Security is a key principle of dxw's development. dxw projects are subject to a
```
Comment (erbridge) [inline rfc-024-use-brakeman.md:14]:
```suggestion
Security is a key principle of dxw's development. dxw projects are normally subject to a
```

File: rfc-024-use-brakeman.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+penetration test before entering the live stage of their lifecycle. It is
+better to address security vulnerabilities in the development  & testing stage,
```
Comment (erbridge) [inline rfc-024-use-brakeman.md:16]:
```suggestion
better to address security vulnerabilities in the development & testing stage,
```

File: rfc-024-use-brakeman.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+## Proposal
+
+Where possible we SHOULD use Brakeman as part of our default CI build. We
+SHOULD ensure a CI build fails if any warnings are raised in Brakeman. We
+SHOULD address warnings where possible, and only add exceptions to Brakeman's
+ignore file after discussion with the development team.
+
+We SHOULD NOT include Brakeman in our default `rake` task. This is not
+recommended by [Brakeman's developers](https://brakemanscanner.org/docs/rake/)
```
Comment (erbridge) [inline rfc-024-use-brakeman.md:32]: I love how vague their reasoning is. Do we know _why_ it's bad to run by default?

Comment (lozette) [reply rfc-024-use-brakeman.md:32]: Yeah, `can cause strange behavior` is 🤔 

When I tried running it in the default Rake task locally it ran, but there were some warnings along the lines of `warning: already initialized constant` caused by Rubocop (standard) loading the same dependencies https://github.com/presidentbeef/brakeman/issues/1424

I don't know if this is serious enough to warrant not adding it to the default Rake task though.

Comment (erbridge) [reply rfc-024-use-brakeman.md:32]: I've always seen it in the default task, myself. Maybe running it at the end would help with that?

Comment (lozette) [reply rfc-024-use-brakeman.md:32]: I've tried moving it around but no dice. There is a recommendation to use the `brakeman-lib` version of the gem instead, if one wants to add it to the default Rake task.

**Pull request merged**